lib/harness/vagrant.js

```diff
@@ -0,0 +1,62 @@
+const SSHConnector = require( "./ssh" );
+const path = require( 'path' );
+const bakerForMacPath = process.platform === 'darwin' ? path.join( require( 'os' ).homedir(), 'Library', 'Baker', 'BakerForMac' ) : undefined;
+const privateKey = process.platform === 'darwin' ? path.join( bakerForMacPath, 'baker_rsa' ) : path.join( boxes, 'baker_rsa' );
+const cmd = require( 'node-cmd' );
+
+class VagrantConnector extends SSHConnector {
+
+    constructor() {
+        super( 'vagrant@', privateKey );
+    }
+
+    async getName( context ) {
+        return context.bakerPath;
+    }
+
+    async tearDown( obj ) {
+        if ( obj && obj.child ) {
+            // 'SIGINT'
+            console.log( `\tTearing down` );
+            obj.child.stdout.removeAllListeners( "data" );
+            obj.child.stdin.write( '\x03' );
+            obj.child.kill();
+        }
+    }
+
+    async sleep( ms ) {
+        return new Promise( resolve => setTimeout( resolve, ms ) );
+    }
+
+    async getSshConfig() {
+        var vagrantData = null;
+
+        cmd.get(
```

Comment:
Going to refactor this to use the built in `child_process.exec()` of Nodejs, to avoid adding one extra dependency.